saleor/checkout/steps.py

```diff
@@ -23,198 +23,161 @@ def __init__(self, request, storage):
                 break
-        if selected_method is None:
+        else:
             # TODO: find cheapest not first
-            selected_method_name, selected_method = delivery_choices[0]
-        self.delivery_method = selected_method
+            selected_method_name, delivery_method = delivery_choices[0]
+        self.delivery_method = delivery_method
         self.forms['delivery'] = DeliveryForm(
-            delivery_choices, request.POST or None,
+            delivery_choices,
+            request.POST or None,
             initial={'method': selected_method_name})
 
-    def __str__(self):
-        return 'shipping-address'
+    def process(self, extra_context=None):
+        context = dict(extra_context or {})
+        context['form'] = self.forms['delivery']
+        return super(ShippingMethodStep, self).process(extra_context=context)
 
     def save(self):
         delivery_form = self.forms['delivery']
-        if self.forms['address'].cleaned_data.get('shipping_same_as_billing'):
-            address = self.billing_address
-        else:
-            address = self.address
-        self.storage['address'] = Address.objects.as_data(address)
-        delivery_method = delivery_form.cleaned_data['method']
-        self.storage['delivery_method'] = delivery_method
-
-    def validate(self):
-        super(ShippingStep, self).validate()
-        if 'delivery_method' not in self.storage:
-            raise InvalidData()
-
-    def forms_are_valid(self):
-        base_forms_are_valid = super(ShippingStep, self).forms_are_valid()
-        delivery_form = self.forms['delivery']
-        if base_forms_are_valid and delivery_form.is_valid():
-            return True
-        return False
+        self.storage['delivery_method'] = delivery_form.cleaned_data['method']
 
     def add_to_order(self, order):
-        self.address.save()
-        order.shipping_method = self.delivery_method.name
-        order.shipping_address = self.address
-        if order.user:
-            User.objects.store_address(order.user, self.address, shipping=True)
+        order.delivery_method = self.delivery_method
 
-    def process(self, extra_context=None):
-        context = dict(extra_context or {})
-        context['delivery_form'] = self.forms['delivery']
-        return super(ShippingStep, self).process(extra_context=context)
+    def validate(self):
+        selected_method_name = self.storage.get('delivery_method')
+        if selected_method_name not in self.valid_delivery_methods:
+            raise InvalidData('Select a valid delivery method')
 
 
 class SummaryStep(BaseCheckoutStep):
     template = 'checkout/summary.html'
     title = _('Summary')
+    forms = {}
+    billing_address = None
 
-    def __init__(self, request, storage, checkout):
-        self.checkout = checkout
+    def __init__(self, request, whole_storage, checkout):
+        self.same_blling_as_shipping = False
+        self.whole_storage = whole_storage
```

Comment:
Why did you changes `self.storage` to `self.whole_storage`? Steps API should be consistent when it's possible